### ISSUE TITLE
Monitoring error with Condor() backend?

### ISSUE THREAD

**Issue by twhyntie** [ganga-devs/ganga#466]
Hello,

So I'm running the Hello World! job on the RAL T1 cluster with the condor backend:

```
j = Job()
j.backend = Condor()
j.backend.getenv = "True"
j.submit()
```
which submits fine (and it's great that the switch to the Condor backend is trivial). I can see the job has been submitted and is running (wahey @alahiff!):

```
Registry Slice: jobs (1 objects)
--------------
    fqid |    status |      name | subjobs |    application |        backend |                             backend.actualCE |                       comment 
-------------------------------------------------------------------------------------------------------------------------------------------------------------
       0 |   running |           |         |     Executable |         Condor |                slot1@lcg1270.gridpp.rl.ac.uk |                               
```

However, I now get the following error spewed out repeatedly:

```
ERROR    Monitoring Error: 'int' object has no attribute 'isdigit'
```

Running with `--debug`:

```
[...]
2016-05-11 11:11:33,994 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::_checkActiveBackends:1094 DEBUG   : calling function _checkActiveBackends
2016-05-11 11:11:33,995 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::__defaultActiveBackendsFunc:938 DEBUG   : __defaultActiveBackendsFunc
2016-05-11 11:11:33,995 GANGA_Update_Thread_JobRegistry_Monitor Registry            ::keys                :463 DEBUG   : keys
2016-05-11 11:11:33,995 GANGA_Update_Thread_JobRegistry_Monitor Registry            ::ids                 :408 DEBUG   : ids
2016-05-11 11:11:33,996 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::__defaultActiveBackendsFunc:944 DEBUG   : Running over fixed_ids: [0]
2016-05-11 11:11:33,996 GANGA_Update_Thread_JobRegistry_Monitor Registry            ::__getitem__         :293 DEBUG   : __getitem__
2016-05-11 11:11:33,996 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::__defaultActiveBackendsFunc:972 DEBUG   : Returning active_backends: {"Condor" : [0, ], }
2016-05-11 11:11:33,996 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::_checkActiveBackends:1104 DEBUG   : Active Backends: {"Condor" : [0, ], }
2016-05-11 11:11:33,997 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::addEntry            :318 DEBUG   : *: backend=Condor, isLocked=1, isOwner=False, joblist=[0], queue=0
2016-05-11 11:11:33,997 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::_checkActiveBackends:1124 DEBUG   : jList: ['0']
2016-05-11 11:11:33,997 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::__monStep           :626 DEBUG   : 

Running runClientCallbacks
2016-05-11 11:11:33,998 GANGA_Update_Thread_JobRegistry_Monitor Local_GangaMC_Service::run                 :562 DEBUG   : Wait Condition
2016-05-11 11:11:34,555 GANGA_Update_Thread_MonitoringWorker_4_1462961429149 Registry            ::_write_access       :793 DEBUG   : _write_access
2016-05-11 11:11:34,555 GANGA_Update_Thread_MonitoringWorker_4_1462961429149 Registry            ::__write_access      :803 DEBUG   : __write_acess
2016-05-11 11:11:34,556 GANGA_Update_Thread_MonitoringWorker_4_1462961429149 Registry            ::_write_access       :793 DEBUG   : _write_access
2016-05-11 11:11:34,556 GANGA_Update_Thread_MonitoringWorker_4_1462961429149 Registry            ::__write_access      :803 DEBUG   : __write_acess
2016-05-11 11:11:34,556 GANGA_Update_Thread_MonitoringWorker_4_1462961429149 FileWorkspace       ::create              :54 DEBUG   : creating /scratch/twhyntie/gangadir/workspace/twhyntie/LocalXML/0/output/
2016-05-11 11:11:34,557 GANGA_Update_Thread_MonitoringWorker_4_1462961429149 Local_GangaMC_Service::_checkBackend       :1049 DEBUG   : Err: 'int' object has no attribute 'isdigit'
ERROR    Monitoring Error: 'int' object has no attribute 'isdigit'
```

The job isn't showing as completed (I've checked with @alahiff and it didn't complete successfully for other reasons, but it is no longer "running"). So is the monitoring failing because of this error?

Thanks, @twhyntie

**Comment by milliams** [comment]
This is being caused by:
```python
try:
    exitLine = lineList[-1]
    exitCode = exitLine.strip().split()[-1]
except IndexError:
    exitCode = -1

if exitCode.isdigit():
    jobStatus = "completed"
else:
    logger.error("Problem extracting exit code from job %s. Line found was '%s'." % (
        jobDict[id].fqid, exitLine))
```
Probably just changing
```python
except IndexError:
    exitCode = -1
```
to
```python
except IndexError:
    exitCode = '-1'
```
would fix it as `'-1'.isdigit()` is `False`. Do you know this code at all @drmarkwslater?

**Comment by drmarkwslater** [comment]
Fix is in PR #482
